appl/include/obj_parser.h

```diff
@@ -86,87 +108,64 @@ obj_t *obj_parse(const char *file_name)
-
-            strtok_s(buffer, " ", &remaining_tokens); // v token
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->v[v_index].x = atof(token);
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->v[v_index].y = atof(token);
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->v[v_index].z = atof(token);
+        char *remaining_tokens;
+        char *token = strtok_s(buffer, " ", &remaining_tokens);
 
+        if (strcmp(token, "v") == 0)
+        {
+            // Get values from buffer and fill struct
+            sscanf_s(remaining_tokens, "%f %f %f", &(obj->v[v_index].x), &(obj->v[v_index].y), &(obj->v[v_index].z));
             v_index++;
         }
 
-        if (strncmp(buffer, "vt ", 3) == 0)
+        else if (strcmp(token, "vt") == 0)
         {
-            char *remaining_tokens;
-            char *token;
-
-            strtok_s(buffer, " ", &remaining_tokens); // vt token
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vt[vt_index].x = atof(token);
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vt[vt_index].y = atof(token);
-
+            // Get values from buffer and fill struct
+            sscanf_s(remaining_tokens, "%f %f", &(obj->vt[vt_index].x), &(obj->vt[vt_index].y));
             vt_index++;
         }
 
-        if (strncmp(buffer, "vn ", 3) == 0)
+        else if (strcmp(token, "vn") == 0)
         {
-            char *remaining_tokens;
-            char *token;
-
-            strtok_s(buffer, " ", &remaining_tokens); // vt token
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vn[vn_index].x = atof(token);
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vn[vn_index].y = atof(token);
-
-            token = strtok_s(NULL, " ", &remaining_tokens);
-            obj->vn[vn_index].z = atof(token);
-
+            // Get values from buffer and fill struct
+            sscanf_s(remaining_tokens, "%f %f %f", &(obj->vn[vn_index].x), &(obj->vn[vn_index].y), &(obj->vn[vn_index].z));
             vn_index++;
         }
 
-        if (strncmp(buffer, "f ", 2) == 0)
+        else if (strcmp(token, "f") == 0)
         {
-            char *remaining_tokens;
-            char *token;
-
-            strtok_s(buffer, " ", &remaining_tokens); // f token
-
             for (int i = 0; i < 3; i++)
             {
-                token = strtok_s(NULL, "/", &remaining_tokens);
-
-                obj->v_info[f_index + i].v_index = atoi(token);
-
-                token = strtok_s(NULL, "/", &remaining_tokens);
-                obj->v_info[f_index + i].vt_index = atoi(token);
-
-                token = strtok_s(NULL, " ", &remaining_tokens);
-                obj->v_info[f_index + i].vn_index = atoi(token);
+                // Get values from buffer and fill struct
+                sscanf_s(remaining_tokens, "%d/%d/%d",
```

Comment:
In general, I wouldn't mix strtok and scanf. For sure it works, but just  having one common way in the code to parse tokens in this scenario could be preferred my opionion.

In particular on this last scenario when parsing faces, it leads to a less readable solution where mixing for/while and pointer aritmetic.

Reply:
> In general, I wouldn't mix strtok and scanf. For sure it works, but just having one common way in the code to parse tokens in this scenario could be preferred my opionion.

I could eliminate the internal while loop with a memove to eliminate the already assigned values each time
What do you think?

> In particular on this last scenario when parsing faces, it leads to a less readable solution where mixing for/while and pointer aritmetic.
Regarding the use of scanf I thought it would be useful. I'll try to find another way
Thanks for feedbacks ;)

Reply:
It is good you try to find other solutions and compare them to look the more efficient.

For this case I think that just strtok can do the job, keeping the code easy to read.

Reply:
I restored the use of stroke, everything should be ok now.